src/ucp/core/ucp_request.inl

```diff
@@ -394,12 +394,14 @@ ucp_request_send_state_reset(ucp_request_t *req,
     case UCP_REQUEST_SEND_PROTO_RNDV_GET:
     case UCP_REQUEST_SEND_PROTO_RNDV_PUT:
     case UCP_REQUEST_SEND_PROTO_ZCOPY_AM:
-        req->send.state.uct_comp.func   = comp_cb;
         req->send.state.uct_comp.count  = 0;
         req->send.state.uct_comp.status = UCS_OK;
         /* Fall through */
     case UCP_REQUEST_SEND_PROTO_BCOPY_AM:
-        req->send.state.dt.offset       = 0;
+        /* Always set completion function to make sure this value is initialized
+         * when doing send fast-forwarding in ucp_request_send_state_ff() */
+        req->send.state.uct_comp.func = comp_cb;
```

Comment:
is it expected to be `NULL` for BCOPY?

Reply:
can we add an assert for this?

Reply:
do ylu mean to add `assert((comp_cb != NULL) || (proto == UCP_REQUEST_SEND_PROTO_BCOPY_AM))`?

Reply:
no I meant ensure that comp_cb is always `NULL` for `UCP_REQUEST_SEND_PROTO_BCOPY_AM`

Reply:
But this is what I meant

Reply:
no, the current assert allows `com_cb` to be non `NULL` for `UCP_REQUEST_SEND_PROTO_BCOPY_AM`